### DISCUSSION TITLE
Query regarding PKCS5_PBKDF2_HMAC() params pass and salt

### DISCUSSION THREAD

**Discussion by vk18anubhav** [openssl/openssl#25384] [Q&A]
What can be the maximum size of parameters pass(password) and salt for function PKCS5_PBKDF2_HMAC() ?

**Comment by kulkarniamit** [comment]
According to [PKCS5_PBKDF2_HMAC](https://docs.openssl.org/master/man3/PKCS5_PBKDF2_HMAC/#description), it doesn't appear like there's a limit on password length or salt length. 

> "_These functions make no assumption regarding the given password. It will simply be treated as a byte sequence._"

There's a lower bound limit on the `iter` count though.

> "_Any iter value less than 1 is invalid; such values will result in failure and raise the PROV_R_INVALID_ITERATION_COUNT error._"

On the other hand, iteration count significantly affects the computational overhead of the function.

#### Additional reading
* [NIST guidelines on password length](https://pages.nist.gov/800-63-4/sp800-63b.html#length)
